deps/v8/src/heap/mark-compact.cc

```diff
@@ -2422,6 +2422,13 @@ void MarkCompactCollector::ClearWeakCollections() {
           table->RemoveEntry(i);
         }
       }
+      // Rehash if more than 25% of the entries are deleted entries.
+      // TODO(jochen): Consider to shrink the fixed array in place.
+      if ((table->NumberOfDeletedElements() << kJSWeakCollectionLoadFactorExp) >
+          table->NumberOfElements()) {
+        HandleScope scope(heap()->isolate());
+        table->Rehash(heap()->isolate()->factory()->undefined_value());
```

Comment:
@jeisinger: Applying the two patches from the 'Reland CL' (https://codereview.chromium.org/1890123002/#ps20001)  produces the same patch as https://github.com/nodejs/node/pull/6398/commits/3e8d7a73aa5d17f070ab17741e48249ea5a9853c. Maybe I am missing something? 

/cc @matthewloring re: https://github.com/nodejs/node/pull/7883